### ISSUE TITLE
request_uri nor compatible with [ipv6] host

### ISSUE THREAD

**Issue by mvorisek** [ledgetech/lua-resty-http#191]
Example URI which is currently failing:

```
http://[::1]/
```

the parsing can be rewritten to:
```
    local m, err = ngx_re_match(uri, [[^(?:(http[s]?):)?//([^\[\]:/\?]+|\[[^\[\]/\?]+\])(?::(\d+))?([^\?]*)\??(.*)]], "jo")
```
but it solves only the parse issue - the `request_uri ` code seems to construct URI directly from the parsed host but it also calls some other methods with the parsed host - should the parsed host contain the `[]` brackets?

**Comment by mvorisek** [comment]
FYI: There is also a PR for the IPv6 host parse https://github.com/ledgetech/lua-resty-http/pull/104

**Comment by pintsized** [comment]
Fixed in #104
